### PR TITLE
Add new cop Style/RedundantParentheses

### PR DESCRIPTION
**Pull request by lumeet**
The cop checks for parentheses that seem to serve neither a syntactic
nor a communicative purpose.

I guess there's room for additional checks but at this point I'm happy with these.

### CODE REVIEW

File: lib/rubocop/cop/style/redundant_parentheses.rb [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+          node.loc.begin && node.loc.begin.is?('('.freeze)
+        end
+
+        def method_call_with_redundant_parentheses?(node)
+          return false unless method_call?(node)
+          return false if range_end?(node)
+
+          send_node, args = method_node_and_args(node)
+
+          send_node.loc.begin || args.size == 0 || square_brackets?(send_node)
```
Comment (bbatsov) [inline lib/rubocop/cop/style/redundant_parentheses.rb]: size == 0 -> empty?

Some day we should have a cop for this... :-)

File: lib/rubocop/cop/style/redundant_parentheses.rb [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        end
+
+        def_node_matcher :literal?, '{str sym int float hash array}'
```
Comment (bbatsov) [inline lib/rubocop/cop/style/redundant_parentheses.rb]: Don't we already have methods like this somewhere? This is definitely not the first cop that uses them.

File: lib/rubocop/cop/style/redundant_parentheses.rb (hunk 1 above)
Comment (lumeet) [inline lib/rubocop/cop/style/redundant_parentheses.rb]: Don't know how it's even possible to still make that mistake? :)

File: lib/rubocop/cop/style/redundant_parentheses.rb (hunk 2 above)
Comment (lumeet) [inline lib/rubocop/cop/style/redundant_parentheses.rb]: Sure, I think I can do some cleanup in other cops, too.

**Comment by bbatsov** [comment]
:+1:

File: lib/rubocop/cop/style/redundant_parentheses.rb (hunk 2 above)
Comment (bbatsov) [inline lib/rubocop/cop/style/redundant_parentheses.rb]: Yeah, that'd be nice.

**Comment by alexdowad** [comment]
:+1:

**Pull request merged**